deploy/deployProjectRegistryZksync.ts

```diff
@@ -0,0 +1,40 @@
+import { Deployer } from "@matterlabs/hardhat-zksync-deploy";
+import * as dotenv from "dotenv";
+import * as ethers from "ethers";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import { utils, Wallet } from "zksync-web3";
+
+dotenv.config();
+
+export default async function (hre: HardhatRuntimeEnvironment) {
+  console.log("Deploying ProjectRegistry contract...");
+
+  // Initialize the wallet
+  const wallet = new Wallet(process.env.DEPLOYER_PRIVATE_KEY ?? "");
+
+  // Create a deployer object and load the artifact(s)
+  const deployer = new Deployer(hre, wallet);
+  const ProjectRegistry = await deployer.loadArtifact("ProjectRegistry");
+
+  // Estimate fee
+  const deploymentFee = await deployer.estimateDeployFee(ProjectRegistry, []);
+ 
+  // OPTIONAL: Deposit funds to L2
+  // Comment this block if you already have funds on zkSync.
+  const depositHandle = await deployer.zkWallet.deposit({
+    to: deployer.zkWallet.address,
+    token: utils.ETH_ADDRESS,
+    amount: deploymentFee.mul(2),
+  });
+  // Wait until the deposit is processed on zkSync
+  await depositHandle.wait();
+
+  const parsedFee = ethers.utils.formatEther(deploymentFee.toString());
+  console.log(`Estimated deployment fee: ${parsedFee} ETH`);
+
+  // Deploy the contract
+  const projectRegistryContract = await deployer.deploy(ProjectRegistry, []);
+
+  // Show the contract info
+  console.log("ProjectRegistry deployed to:", projectRegistryContract.address);
+}
```

Comment:
Oh that's not a lot of work to get this rolled out !
Would it be possible to add this to our deployProjectRegistry script instead and that way we have it wrapped it in place (given that most of the code is duplicate )

Reply:
From the looks of this - it seems to be quite a bit different of a deployment script than the default evm `deployProjectRegistry` script. 

You can see more in the zk-era docs here: https://era.zksync.io/docs/dev/building-on-zksync/hello-world.html#initializing-the-project-deploying-a-smart-contract

If so - should it still be merged into a single file?

Reply:
This is a good question @hmrtn I am putting in one file for now and we could refactor. There is a lot of repeated "boilerplate" code.